src/main/java/org/openhab/binding/zigbee/converter/ZigBeeConverterColorColor.java

```diff
@@ -106,72 +280,206 @@ public void run() {
+                        updateBrightness(brightness);
+                    }
+                } else if (attribute.getCluster().getId() == ZclColorControlCluster.CLUSTER_ID) {
+                    if (attribute.getId() == ZclColorControlCluster.ATTR_CURRENTHUE) {
+                        Integer value = (Integer) attribute.getLastValue();
+                        float hue = value * 360.0f / 254.0f;
+                        if (hue != lastHue) {
+                            lastHue = hue;
+                            hueChanged = true;
+                        }
+                    } else if (attribute.getId() == ZclColorControlCluster.ATTR_CURRENTSATURATION) {
+                        Integer value = (Integer) attribute.getLastValue();
+                        float saturation = value * 100.0f / 254.0f;
+                        if (saturation != lastSaturation) {
+                            lastSaturation = saturation;
+                            saturationChanged = true;
+                        }
+                    } else if (attribute.getId() == ZclColorControlCluster.ATTR_CURRENTX) {
+                        Integer value = (Integer) attribute.getLastValue();
+                        float x = value / 65536.0f;
+                        if (x != lastX) {
+                            lastX = x;
+                            xChanged = true;
+                        }
+                    } else if (attribute.getId() == ZclColorControlCluster.ATTR_CURRENTY) {
+                        Integer value = (Integer) attribute.getLastValue();
+                        float y = value / 65536.0f;
+                        if (y != lastY) {
+                            lastY = y;
+                            yChanged = true;
+                        }
+                    }
                 }
-            }
-            if (attribute.getId() == ZclColorControlCluster.ATTR_CURRENTSATURATION) {
-                Integer value = (Integer) attribute.getLastValue();
-                if (value != null) {
-                    PercentType saturation = new PercentType(value * 100 / 254);
-                    currentHSB = new HSBType(currentHSB.getHue(), saturation, currentHSB.getBrightness());
-                    updateChannelState(currentHSB);
+
+                if (hueChanged || saturationChanged || xChanged || yChanged) {
+                    if (colorUpdateTimer != null) {
+                        colorUpdateTimer.cancel(true);
+                        colorUpdateTimer = null;
+                    }
+
+                    if (hueChanged && saturationChanged) {
+                        updateColorHSB();
+                    } else if (xChanged && yChanged) {
+                        updateColorXY();
+                    } else {
+                        // Wait some time and update anyway if only one attribute in each pair is updated
+                        colorUpdateTimer = colorUpdateScheduler.schedule(new Runnable() {
```

Comment:
This will still end up with the thread being created (I think). You probably ought to cancel the timer - I think this could be done.

I would think adding the following code just above here at line 451 should do it. This way the timer is cancelled, and will only ever run if we end up in the default statement.

```
if(colorUpdateTimer != null) {
    colorUpdateTimer.cancel();
    colorUpdateTimer = null;
}
```

Reply:
Hi Chris, it is already there, exactly between lines 447 and 450 ::)

Reply:
Hmmm - my bad! Thanks.